### PR TITLE
schedulers: support hot read peer scheduling

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What is changed and how it works?

Support transfer follower peer for hot read scheduling

### Check List

Tests

- Unit test

### Test Result

sysbench --config-file=./sysbench.config oltp_read_only --tables=4 --table-size=43000000--create-secondary=true run

image

image

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/mock/mockcluster/mockcluster.go [hunk 1]
```diff
@@ -353,6 +353,30 @@ func (mc *Cluster) AddRegionWithReadInfo(
 	return items
 }
 
+// AddRegionWithPeerReadInfo adds region with specified peer read info.
+func (mc *Cluster) AddRegionWithPeerReadInfo(regionID, leaderID, targetStoreID, readBytes, readKeys, reportInterval uint64,
+	followerIds []uint64, filledNums ...int) []*statistics.HotPeerStat {
+	r := mc.newMockRegionInfo(regionID, leaderID, followerIds...)
+	r = r.Clone(core.SetReadBytes(readBytes))
```
Comment (rleungx) [inline pkg/mock/mockcluster/mockcluster.go:360]: No need to clone for 3 times.

File: server/schedulers/hot_test.go [hunk 2]
```diff
@@ -1108,6 +1117,21 @@ func addRegionLeaderReadInfo(tc *mockcluster.Cluster, regions []testRegionInfo)
 	}
 }
 
+func addRegionPeerReadInfo(tc *mockcluster.Cluster, regions []testRegionPeerInfo) {
+	addFunc := tc.AddRegionWithPeerReadInfo
+	reportIntervalSecs := statistics.ReadReportInterval
+	for _, r := range regions {
+		addFunc(
```
Comment (rleungx) [inline server/schedulers/hot_test.go:1124]: Why not directly use `tc.AddRegionWithPeerReadInfo`

File: server/schedulers/hot_test.go [hunk 3]
```diff
@@ -845,7 +845,7 @@ func (s *testHotReadRegionSchedulerSuite) TestWithPendingInfluence(c *C) {
 			// store byte/key rate (min, max): (6.6, 7.1) | 6.1 | (6, 6.5) | 5
 
 			op2 := hb.Schedule(tc)[0]
-			testutil.CheckTransferPeerWithLeaderTransfer(c, op2, operator.OpHotRegion, 1, 4)
+			testutil.CheckTransferPeer(c, op2, operator.OpHotRegion, 1, 4)
```
Comment (rleungx) [inline server/schedulers/hot_test.go:848]: Why

File: server/cluster/coordinator.go [hunk 4]
```diff
@@ -556,6 +556,17 @@ func (c *coordinator) collectHotSpotMetrics() {
 			hotSpotStatusGauge.WithLabelValues(storeAddress, storeLabel, "hot_read_region_as_leader").Set(0)
 		}
 
+		stat, ok = status.AsPeer[storeID]
```
Comment (lhy1024) [inline server/cluster/coordinator.go:559]: metrics also need to be updated

File: server/schedulers/hot_test.go (hunk 3 above)
Comment (Yisaer) [reply server/schedulers/hot_test.go:848]: As we have supported hot read peer scheduling, the hot read scheduling may not always transfer leader peer.

**Review by rleungx — APPROVED** [review]

File: server/cluster/coordinator.go (hunk 4 above)
Comment (Yisaer) [reply server/cluster/coordinator.go:559]: updated.

**Review by nolouch — COMMENTED** [review]
Could you plz share some test results in here?

**Comment by Yisaer** [comment]
> Could you plz share some test results in here?

I will upload test results later

**Comment by lhy1024** [comment]
LGTM, I will approve after you upload test results.

**Review by lhy1024 — APPROVED** [review]
LGTM

**Pull request merged**